clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2296,6 +2315,18 @@ void CodeGenModule::EmitDeferred() {
     // Otherwise, emit the definition and move on to the next one.
     EmitGlobalDefinition(D, GV);
 
+    if (LangOpts.SYCLIsDevice) {
+      // If there are any aliases deferred for this, emit those now.
+      for (auto it = DeferredAliases.begin(); it != DeferredAliases.end();
+           ++it) {
```

Comment:
I can't do range for here since I am invalidating the iterator within the body.   Which made me realize I had a bug in my previous for loop, which I have fixed.